grpc.go

```diff
@@ -579,49 +674,45 @@ func (a *agentGRPC) CreateContainer(ctx context.Context, req *pb.CreateContainer
 		return emptyResp, err
 	}
 
-	// Convert the OCI specification into a libcontainer configuration.
-	config, err := specconv.CreateLibcontainerConfig(&specconv.CreateOpts{
-		CgroupName:   req.ContainerId,
-		NoNewKeyring: true,
-		Spec:         ociSpec,
-		NoPivotRoot:  a.sandbox.noPivotRoot,
-	})
+	// Add any custom OCI hooks to the spec
+	err = addHooks(ociSpec)
```

Comment:
The hooks are static afaics. So it is more efficient to just scan once during agent startup, and do the OCI spec/hook dance only when there are real hooks to add.